pkg/crc/machine/status.go

```diff
@@ -57,13 +64,18 @@ func (client *client) Status() (*types.ClusterStatusResult, error) {
 		DiskUse:   diskUse,
 		DiskSize:  diskSize,
 	}
-	if vm.bundle.IsOpenShift() {
+	switch {
+	case vm.bundle.IsPodman():
+		clusterStatusResult.PodmanVersion = vm.bundle.GetPodmanVersion()
+		clusterStatusResult.Preset = preset.Podman
+	case vm.bundle.IsMicroshift():
+		clusterStatusResult.OpenshiftStatus = getMicroShiftStatus(vm)
+		clusterStatusResult.OpenshiftVersion = vm.bundle.GetOpenshiftVersion()
+		clusterStatusResult.Preset = preset.Microshift
+	default:
```

Comment:
The code is starting to get quite repetitive, can you first create a `clusterStatusResult` with with the fields Preset and xxxVersion set. Then you can use this variable in the 2 code paths (cluster is stopped or running) and only set the correct Status  field. This can even be moved to different helpers for improved readabililty.